ivy/functional/frontends/torch/linalg.py

```diff
@@ -94,3 +94,33 @@ def svd(A, /, *, full_matrices=True, driver=None, out=None):
 def svdvals(A, *, driver=None, out=None):
     # TODO: add handling for driver
     return ivy.svdvals(A, out=out)
+
+
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes({"1.11.0 and below": ("float16", "bfloat16")}, "torch")
+def tensorinv(input, ind=2):
+    input_shape = ivy.shape(input)
+    if ind > 0:
+        shape_ind_end = input_shape[:ind]
+        shape_ind_start = input_shape[ind:]
+        prod_ind_end = 1
+        prod_ind_start = 1
+        for i in shape_ind_start:
+            prod_ind_start *= i
+        for j in shape_ind_end:
+            prod_ind_end *= j
+        if prod_ind_end == prod_ind_start:
+            inverse_shape = shape_ind_start + shape_ind_end
+            input = ivy.reshape(input, shape=(prod_ind_end, -1))
+            inverse_shape_tuple = tuple([*inverse_shape])
+            if len(ivy.shape(input)) > 1:
+                inverse_tensor = ivy.inv(input)
+            else:
+                return ivy.reshape(input, shape=inverse_shape_tuple)
+        else:
+            raise ValueError(
+                            "Tensor shape must satisfy prod"
+                            "(A.shape[:ind]) == prod(A.shape[ind:])")
+    else:
```

Comment:
You should also check if 'A' is invertible and raise an error if it's not.

Reply:
Could you please check if it is done properly?